components/dashboard/src/prebuilds/list/PrebuildTable.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { FC } from "react";
+import { Table, TableBody, TableHead, TableHeader, TableRow } from "@podkit/tables/Table";
+import { RepositoryListItem } from "./PrebuildListItem";
+import { LoadingButton } from "@podkit/buttons/LoadingButton";
+import { TextMuted } from "@podkit/typography/TextMuted";
+import { Subheading } from "@podkit/typography/Headings";
+import { cn } from "@podkit/lib/cn";
+import { LoadingState } from "@podkit/loading/LoadingState";
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@podkit/select/Select";
+import { Prebuild } from "@gitpod/public-api/lib/gitpod/v1/prebuild_pb";
+import { Filter, SortField, StatusOption } from "./PrebuildList";
+import { SortCallback, SortableTableHead, TableSortOrder } from "@podkit/tables/SortableTable";
+import ConfigurationDropdown from "./ConfigurationInput";
+
+type Props = {
+    prebuilds: Prebuild[];
+    filter?: Filter;
+    sort: {
+        sortBy: SortField;
+        sortOrder: TableSortOrder;
+    };
+    hasNextPage: boolean;
+    hasMoreThanOnePage: boolean;
+    isSearching: boolean;
+    isFetchingNextPage: boolean;
+    onFilterChange: (val: Filter) => void;
+    onLoadNextPage: () => void;
+    onSort: (columnName: SortField, direction: TableSortOrder) => void;
+};
+export const PrebuildsTable: FC<Props> = ({
+    prebuilds,
+    hasNextPage,
+    hasMoreThanOnePage,
+    isSearching,
+    isFetchingNextPage,
+    filter,
+    sort,
+    onFilterChange,
+    onLoadNextPage,
+    onSort,
+}) => {
+    return (
+        <>
+            {/* Search/Filter bar */}
+            <div className="flex flex-col-reverse md:flex-row flex-wrap justify-between items-center gap-2">
+                <div className="flex flex-row flex-wrap gap-2 items-center w-full md:w-auto">
+                    <ConfigurationDropdown
+                        selectedConfigurationId={filter?.configurationId}
+                        onChange={(configurationId) => {
+                            if (!configurationId) {
+                                // If the reset item is selected, we reset the filter
+                                onFilterChange({ ...filter, configurationId: undefined });
```

Comment:
Code inside if and outside of if is the same? (PS. It missed an `return` even they're not the same

Reply:
Thank you for the missing return. Did this originally because `configurationId` could be `""` here and not explicitly `undefined`, but it should work as it is still falsy with the rest of the code.